src/lsptoolshost/dotnetRuntimeExtensionResolver.ts

```diff
@@ -151,39 +150,40 @@ export class DotnetRuntimeExtensionResolver implements IHostExecutableResolver {
             }
 
             const coreRuntimeVersions = dotnetInfo.Runtimes['Microsoft.NETCore.App'];
-            let foundRuntimeVersion = false;
-            for (const version of coreRuntimeVersions) {
+            let matchingRuntime: RuntimeInfo | undefined = undefined;
+            for (const runtime of coreRuntimeVersions) {
                 // We consider a match if the runtime is greater than or equal to the required version since we roll forward.
-                if (semver.gt(version, requiredRuntimeVersion)) {
-                    foundRuntimeVersion = true;
+                if (semver.gt(runtime.Version, requiredRuntimeVersion)) {
+                    matchingRuntime = runtime;
                     break;
                 }
             }
 
-            if (!foundRuntimeVersion) {
+            if (!matchingRuntime) {
                 throw new Error(
                     `No compatible .NET runtime found. Minimum required version is ${this.minimumDotnetRuntimeVersion}.`
                 );
             }
 
-            // Find the location of the dotnet on path.
-            const command = this.platformInfo.isWindows() ? 'where' : 'which';
-            const whereOutput = await promisify(exec)(`${command} dotnet`);
-            if (!whereOutput.stdout) {
-                throw new Error(`Unable to find dotnet from ${command}.`);
-            }
-
-            // There could be multiple paths output from where.  Take the first since that is what we used to run dotnet --info.
-            const path = whereOutput.stdout.trim().replace(/\r/gm, '').split('\n')[0];
-            if (!existsSync(path)) {
-                throw new Error(`dotnet path does not exist: ${path}`);
+            // The .NET install layout is a well known structure on all platforms.
+            // See https://github.com/dotnet/designs/blob/main/accepted/2020/install-locations.md#net-core-install-layout
+            //
+            // Therefore we know that the runtime path is always in <install root>/shared/<runtime name>
+            // and the dotnet executable is always at <install root>/dotnet(.exe).
+            //
+            // Since dotnet --list-runtimes will always use the real assembly path to output the runtime folder (no symlinks!)
+            // we know the dotnet executable will be two folders up in the install root.
+            const runtimeFolderPath = matchingRuntime.Path;
+            const installFolder = path.dirname(path.dirname(runtimeFolderPath));
+            const dotnetExecutablePath = path.join(installFolder, this.getDotnetExecutableName());
+            if (!existsSync(dotnetExecutablePath)) {
+                throw new Error(
+                    `dotnet executable path does not exist: ${dotnetExecutablePath}, dotnet installation may be corrupt.`
+                );
             }
 
             this.channel.appendLine(`Using dotnet configured on PATH`);
```

Comment:
Does this message need to change? Or should it now log the path we created?

Reply:
the path will be logged in the caller anway, and we're still using the dotnet from the path, so think this is fine.